### PR TITLE
New node post. DO NOT MERGE UNTIL TOMORROW!

### PR DESCRIPTION
**Pull request by rdegges**
New post.

### CODE REVIEW

File: _source/_posts/2018-04-24-simple-node-authentication.md [hunk 1]
```diff
@@ -0,0 +1,732 @@
...
+
+## Create a Simple Node Application
+
+Now that you have Node installed, let's get started writing an app.
+
+**PS**: If you want to skip the tutorial and jump straight to the finished product you can check out this sample app on GitHub here: https://github.com/oktadeveloper/okta-node-express-example
+
+In the Node ecosystem, the most popular web server is [`Express`](https://expressjs.com/). Luckily for you, the Express team has also created a nifty [command line tool](https://expressjs.com/en/starter/generator.html) which helps bootstrap Express web applications. You can always customize the bootstrapped project `express-generator` generates, but let's start with that to get up and running quickly.
+
+```bash
```
Comment (mraible) [inline _source/_posts/2018-04-24-simple-node-authentication.md]: I'd remove the `bmk@redmbp` lines in the code below...

File: _source/_posts/2018-04-24-simple-node-authentication.md [hunk 2]
```diff
@@ -0,0 +1,732 @@
...
+bmk@redmbp ~/code
+$ npm i -g express-generator
```
Comment (mraible) [inline _source/_posts/2018-04-24-simple-node-authentication.md]: This should be versioned - `npm i -g express-generator@4.16.0`

File: _source/_posts/2018-04-24-simple-node-authentication.md [hunk 3]
```diff
@@ -0,0 +1,732 @@
...
+    "no-var": ["error"],
+    "prefer-const": ["error"]
+  }
+}
+```
+
+Now you need to install a few dev dependencies (the `-D` flag is for dev dependencies).
+
+```bash
+  npm i -D eslint eslint-config-standard eslint-plugin-standard eslint-plugin-promise eslint-plugin-import eslint-plugin-node
```
Comment (mraible) [inline _source/_posts/2018-04-24-simple-node-authentication.md]: I'd say these should be versioned, but I'm not sure it's as important for a linter.

File: _source/_posts/2018-04-24-simple-node-authentication.md [hunk 4]
```diff
@@ -0,0 +1,732 @@
...
+
+## Add Authentication With Okta
+
+The Express generator created a skeleton for you to get started. Now, let's add authentication. Traditionally, there is a lot to think about when it comes to creating a safe, secure application. For example, what are best practices for storing hashed passwords? Do these best practices change often? Should you use multi-factor authentication? To say there's a lot of research to do this right is an understatement.
+
+Luckily, Okta makes this really easy by offloading authentication and letting you focus on what makes your app unique. To get started [sign up for a free Okta Developer account](https://developer.okta.com/signup/).
+
+You'll get a link to your own Okta Org URL which you'll use to sign in from now on, so you should save it for later. It'll look something like `https://dev-123456.oktapreview.com`. In order to log in, check your email for a temporary password.
+
+After logging into the dev console, click on the `Applications` tab, then click `Add Application`. For this example, you should use the `Web` option.
```
Comment (mraible) [inline _source/_posts/2018-04-24-simple-node-authentication.md]: I'd recommend bolding these. That's what I do in my posts.

File: _source/_posts/2018-04-24-simple-node-authentication.md [hunk 5]
```diff
@@ -0,0 +1,732 @@
...
++module.exports = { app, oidc }
+```
+
+The Okta middleware adds a `logout` function to the request object, so you can log users out with a call to `req.logout()`. That was added above as a route, so you just need a link to `/logout` in order to sign out, which I'll show you a bit later.
+
+For this example you're not going to be using the `users` route that came with the skeleton either, so you can remove that by running `git rm routes/users.js`.
+
+You might be wondering where all the `process.env` variables are coming from. It's a best practice not to store sensitive information in source control, so you can instead use `dotenv`. This lets you keep things like your client secret in a file that you ignore from source control. It also lets you define different variables depending on whether you're running in development or production.
+
+For now, install the package with `npm i dotenv`, then create a `.env` file in the root of the project.
```
Comment (mraible) [inline _source/_posts/2018-04-24-simple-node-authentication.md]: Should this`npm i` be versioned?

File: _source/_posts/2018-04-24-simple-node-authentication.md [hunk 6]
```diff
@@ -0,0 +1,732 @@
...
++  server.listen(port)
++})
++oidc.on('error', onError)
+ server.on('error', onError)
+ server.on('listening', onListening)
+```
+
+## Add a Dashboard and Login Page
+
+Let'screate a private dashboard that's only accessible for logged in users. In  `app.js` you already added the reference to your dashboard's router, so you'll need to add that in `routes/dashboard.js`.
```
Comment (mraible) [inline _source/_posts/2018-04-24-simple-node-authentication.md]: Needs space after "Let's"

File: _source/_posts/2018-04-24-simple-node-authentication.md [hunk 7]
```diff
@@ -0,0 +1,732 @@
...
+    title: 'Dashboard',
+    descriptionList,
+    userinfo,
+  })
+})
+
+module.exports = router
+```
+
+I added `lodash`, so you'll need to install that as a dependency with `npm i lodash`. You'll also want to add a view to describe how to render the data in `views/dashboard.hbs`.
```
Comment (mraible) [inline _source/_posts/2018-04-24-simple-node-authentication.md]: Does lodash need to be versioned?

File: _source/_posts/2018-04-24-simple-node-authentication.md [hunk 8]
```diff
@@ -0,0 +1,732 @@
...
+
+{% img blog/simple-node-authentication/simple-node-auth-dashboard.png alt:"Simple Node Auth Dashboard" width:"600" %}{: .center-image }
+
+## Add User Registration
+
+You should have a functional app now with authentication and protected pages, created with only a few additions to the basic skeleton app. You don't have any way to add new users though.
+
+Let's create a simple registration page. For this, we'll be using the [Okta Node.js SDK](https://github.com/okta/okta-sdk-nodejs), which you can install with `npm i @okta/okta-sdk-nodejs`.
+
+You'll also need an API token from Okta to be able to create a user. After logging in to your developer console, select API -> Tokens from the menu.
```
Comment (mraible) [inline _source/_posts/2018-04-24-simple-node-authentication.md]: These links should match the ones above. Above uses backticks, but I recommend bold.

File: _source/_posts/2018-04-24-simple-node-authentication.md [hunk 9]
```diff
@@ -0,0 +1,732 @@
...
+
+Now you have a fully functioning web application with user registration, authentication, and credential recovery, without having to has a single password or manage a single user. You can have your cake, and eat it too!
+
+OK, the cake is a lie, but you can now focus on what makes your application great and unique while offloading the complications of user management to Okta. If you'd like to learn more about Node, Okta, authentication, or application security, I'd encourage you to check out the rest of the Okta developer blog, and specifically these great resources:
+
+* [Our Node SDK Documentation](https://developer.okta.com/code/nodejs/)
+* [Okta OIDC Middleware Documentation](https://github.com/okta/okta-oidc-js/tree/master/packages/oidc-middleware)
+* [Build User Registration with Node, React, and Okta](https://developer.okta.com/blog/2018/02/06/build-user-registration-with-node-react-and-okta)
+* [Build a Basic CRUD App with Vue.js and Node](https://developer.okta.com/blog/2018/02/15/build-crud-app-vuejs-node)
+* [What the Heck is OAuth?](https://developer.okta.com/blog/2017/06/21/what-the-heck-is-oauth)
```
Comment (mraible) [inline _source/_posts/2018-04-24-simple-node-authentication.md]: Blog posts should be relative links.

File: _source/_posts/2018-04-24-simple-node-authentication.md (hunk 5 above)
Comment (rdegges) [reply _source/_posts/2018-04-24-simple-node-authentication.md]: This one is fine to skip. It never changes =0

File: _source/_posts/2018-04-24-simple-node-authentication.md (hunk 3 above)
Comment (rdegges) [reply _source/_posts/2018-04-24-simple-node-authentication.md]: Yah -- good point =0

File: _source/_posts/2018-04-24-simple-node-authentication.md [hunk 10]
```diff
@@ -0,0 +1,732 @@
...
+- "Want to build standards-based, simple, straightforward authentication into your next Node app? Read our new guide:"
+---
+
+Authenticating users for Node.js/Express.js web apps can be difficult. You have to set up a database, define a user schema (or use something more flexible like NoSQL), write code to handle password hashing, etc. It's annoying.
+
+Using Okta's API service, however, you can easily register and log in users to your Node website using our OpenID Connect integration.
+
+## Set Up Your Node Environment
+
+If you're new to Node and don't already have it installed, you'll need to do that first. To keep things simple, let's assume a \*nix-based environment (e.g. Mac, Linux, or [Linux on Windows](https://docs.microsoft.com/en-us/windows/wsl/about)).
```
[the remark this answers is not in the document]
Comment (rdegges) [reply _source/_posts/2018-04-24-simple-node-authentication.md]: fixed.

File: _source/_posts/2018-04-24-simple-node-authentication.md (hunk 7 above)
Comment (rdegges) [reply _source/_posts/2018-04-24-simple-node-authentication.md]: done

**Pull request merged**